src/file_resolver.rs

```diff
@@ -199,49 +190,69 @@ impl FileResolver {
                 let path = base.join(&path);
 
                 if let Some(file) = self.try_file(filename, &path, *import_no)? {
+                    // No ambiguity possible, so just return
                     return Ok(file);
                 }
             }
 
             return Err(format!("file not found '{}'", filename.to_string_lossy()));
         }
 
-        if let Some(file) = self.try_file(filename, &path, None)? {
-            return Ok(file);
-        } else if path.is_absolute() {
-            return Err(format!("file not found '{}'", filename.to_string_lossy()));
+        if parent.is_none() {
+            if let Some(file) = self.try_file(filename, &path, None)? {
+                return Ok(file);
+            } else if path.is_absolute() {
+                return Err(format!("file not found '{}'", filename.to_string_lossy()));
+            }
```

Comment:
We only want to use the actual filename (i.e., not try it against import maps) when `parent` is `None`. This is because `parent.is_none()` if and only if the file was passed in from command line (i.e., is not an import).